docs/src/pages/presets/writing-presets/index.md

```diff
@@ -0,0 +1,120 @@
+---
+id: 'writing-presets'
+title: 'Writing Presets'
+---
+
+[Storybook presets](../introduction/) are grouped collections of `babel`, `webpack`, and `addons` configurations that support specific use cases in Storybook, such as typescript or MDX support.
+
+This doc covers the [presets API](#presets-api) and how to use the presets mechanism for [advanced configuration](#advanced-configuration).
+
+## Presets API
+
+A preset is a set of hooks that is called by Storybook on initialization and can override configurations for `babel`, `webpack`, `addons`, and `entries`.
```

Comment:
Is there any thought or plans for allowing presets to handle `addDecorator` or `addParameters` calls? My use case is a company with many separate apps, but all fall under the same stack, and my team is in charge of the stack. We'd really like to be able to have a single preset/config that all of our different teams can use (for consistency's sake), and then when we make updates (like when storybook-docs drops) we can update the preset/config in 1 spot, and everyone gets the wonderful changes.

I've tried putting most of my shared configuration into a file and importing it into the .storybook/config.js file, but it appears like it isn't "run" for storybook. Even passing the `addDecorator` function from .storybook/config.js into the shared configuration file and calling it in there didn't work.

Reply:
You should file that as an issue. I'd expect it to work in Storybook. I also think that could be a cool feature to add to presets.